src/main/kotlin/it/pagopa/ecommerce/payment/requests/services/CartService.kt

```diff
@@ -25,28 +36,47 @@ class CartService {
          * The carts redirect url is composed as follow
          * {host}/{fiscalCode}{noticeNumber}
          */
-        const val CARTS_REDIRECT_URL_FORMAT: String = "%s/%s%s"
-    }
+        const val CARTS_REDIRECT_URL_FORMAT: String = "%s/carts/%s%s"
 
-    @Value("\${checkout.url}")
-    lateinit var checkoutUrl: String
+        const val MAX_ALLOWED_PAYMENT_NOTICES: Int = 1
+    }
 
     /*
      * Process input cartRequestDto:
      * - 1 payment notice is present -> redirect response is given to the checkout location
      * - 2 or more payment notices are present -> error response
      */
-    fun processCart(cartRequestDto: CartRequestDto): String {
+    suspend fun processCart(cartRequestDto: CartRequestDto): String {
         val paymentsNotices = cartRequestDto.paymentNotices
         val receivedNotices = paymentsNotices.size
         logger.info("Received [$receivedNotices] payment notices")
-        return if (receivedNotices == 1) {
+
+        return if (receivedNotices == CartServiceConstants.MAX_ALLOWED_PAYMENT_NOTICES) {
             val paymentNotice = paymentsNotices[0]
-            CartServiceConstants.CARTS_REDIRECT_URL_FORMAT.format(
+            val paymentInfos = paymentsNotices.map {
+                PaymentInfo(RptId(it.fiscalCode + it.noticeNumber), it.description, it.amount, it.companyName)
+            }
+
+            val returnUrl = CartServiceConstants.CARTS_REDIRECT_URL_FORMAT.format(
                 checkoutUrl,
                 paymentNotice.fiscalCode,
                 paymentNotice.noticeNumber
```

Comment:
the redirect url must be /carts/cartId where cartId is the UUID of the redis created cart. Change the string template and put here the cart id